airflow/api_connexion/schemas/task_instance_schema.py

```diff
@@ -33,6 +33,14 @@
 from airflow.utils.state import State
 
 
+class _ExecutorConfigField(fields.String):
+    def _serialize(self, value, attr, obj, **kwargs):
+        try:
+            return super()._serialize(value, attr, obj, **kwargs)
+        except Exception:
```

Comment:
What exceptions are possible? This `except` seems much too broad to me.

Reply:
The `executor_config` in case of kubernetes can have values serialized and stored in one version to be picked up later by another kubernetes version in case of Airflow upgrade with old task instances. We noticed `AttributeError` in our case as per the issue but if the serialization fails due to other cases we just want to return `{}` so that other task instances are not affected. There could be other exceptions here as well. PR is similar to https://github.com/apache/airflow/pull/24117

Reply:
@uranusjr curious what's problematic with being very broad.... the goal i guess is to not fail the whole request just because one field fails to serialize...?

Reply:
I think the problem is mostly that we have no way to communicate the failed serialisation if we swallow everything here.

Reply:
@uranusjr  how would you recommend we communicate failed serialization?  re what exceptions it could be... we should no longer have an issue with k8s pod objects.  and the only thing i know of that is ever put in exec config is pod object.  so i am not sure we have any way to know what other exceptions we might expect.

Reply:
It's generally bad form to catch all Exceptions because if you get an exception didn't expect it would be better for things to fail in a _big_ way instead of a controlled way.  A lot of linters explicitly call this out. I think it's reasonable to catch the exceptions we expect to catch given the knowledge we have instead of a broad Exception.